web/app/charts/SankeyEditor.js

```diff
@@ -322,6 +322,91 @@ bluewave.charts.SankeyEditor = function(parent, config) {
         button.hospital.enable();
     };
 
+  //**************************************************************************
+  //** getPreviousNodeValue
+  //**************************************************************************
+    var getPreviousNodeValue = function(previousNodeId){
+        var node = nodes[previousNodeId];
+        var inputs = node.inputs;
+        var quantity = 0;
+        for(var k in inputs) {
+            if(inputs.hasOwnProperty(k)) {
+                var v = quantities[k + "->" + previousNodeId];
+                quantity = quantity + v;
+            }
+        }
+        if(quantity == 0){
+            quantity = 1;
+        }
+        return quantity
+    }
+
+  //**************************************************************************
+  //** auditNodes
+  //**************************************************************************
+    var auditNodes = function(){
+        for (var key in nodes) {
+             var node = nodes[key];
+             var inputs = node.inputs;
+             var outputs = node.outputs;
+             var inputQuantity = 0;
+             var outputQuantity = 0;
+             for(var k in inputs) {
+                if(inputs.hasOwnProperty(k)) {
+                    var n = nodes[k];
+                    var v = quantities[k + "->" + key];
+                    inputQuantity = inputQuantity + v;
+                }
+
+            }
+            for(var k in outputs) {
+                if(outputs.hasOwnProperty(k)) {
+                    var n = nodes[k];
+                    var v = quantities[key + "->" + k];
+                    outputQuantity = outputQuantity + v;
+                }
+            }
+            if(node.type == "distributor" && inputQuantity != outputQuantity) {
+                node.style.color = "red";
+            } else {
+                node.style.color = "black";
+            }
+        }
+        auditLinkages();
+    }
+
+  //**************************************************************************
+  //** auditLinkages
+  //**************************************************************************
+    var auditLinkages = function(){
+        var linkages = document.getElementsByClassName("connection");
+        for(var item of linkages) {
+            var classNames = item.classList;
+            var nodeList = [];
+            for(var name of classNames) {
+                if(name.includes("node")){
+                    nodeList.push(name);
+                }
+            }
+            var path = item.children[0];
+            checkLinkage(nodeList, path);
+        }
+    }
+
+  //**************************************************************************
+  //** checkLinkage
+  //**************************************************************************
+    var checkLinkage = function(nodeList, path){
+        nodeIn = nodes[nodeList[0].replace("node_in_node_", "")];

```

Comment:
Unfortunately this is not going to work as expected. Drawflow is a little inconsistant with the classnames - sometimes they use hyphens vs unscore. See around line 194 for an alternative

Reply:
I've fixed this using your alternative as the basis.